### PR TITLE
Exposing getNonManifoldEdgeMap, adding mesh volume

### PR DESCRIPTION
**Pull request by katarinagresova**
Exposing `getNonManifoldEdgeMap()` and adding `volumne()`.

### CODE REVIEW

File: geometry/mesh.hpp [hunk 1]
```diff
@@ -230,6 +230,10 @@ struct Mesh {
     /** Calculate face barycenter.
      */
     math::Point3 barycenter(const Face &face) const;
+
+    /** Calculate volume of a mesh.
```
Comment (tomas2211) [inline geometry/mesh.hpp:234]: I'd add a note that the formula only works on watertight meshes. (may seem obvious, but...)

```suggestion
    /** Calculate volume of a watertight mesh.
```

**Review by tomas2211 — APPROVED** [review]
I like the volume formula :smiley:

File: geometry/mesh.cpp [hunk 2]
```diff
@@ -171,4 +171,33 @@ math::Point3 Mesh::barycenter(const Face &face) const
...
+double signedVolumeOfTriangle(const math::Point3& p1, const math::Point3& p2, const math::Point3& p3)
+{
+    double v321 = p3[0]*p2[1]*p1[2];
+    double v231 = p2[0]*p3[1]*p1[2];
+    double v312 = p3[0]*p1[1]*p2[2];
+    double v132 = p1[0]*p3[1]*p2[2];
+    double v213 = p2[0]*p1[1]*p3[2];
+    double v123 = p1[0]*p2[1]*p3[2];
+
+    return (1.0f/6.0f)*(-v321 + v231 + v312 - v132 - v213 + v123);
```
Comment (vaclavblazek) [inline geometry/mesh.cpp:185]: `1.0f/6.0 ` is a `float` that is less precise than `double` `1.0/6.0`. Please, use `1.0/6.0`.

Rationale:
Generated assembly uses these constants that are used to multiply the number (via `mulsd` instruction, https://www.felixcloutier.com/x86/mulsd)

for double 1.0/6.0 
```
        .long   1431655765
```        
for float 1.0f/6.0f:
```
        .long   1610612736
```        
(these are "integer" representation of the fp numbers)

The difference in these constants is:
code:
```
auto f(1.0f/6.0f);
auto d(1.0/6.0);
auto diff(d-f);
    std::cout << "d: " << std::setprecision(20) << std::scientific << d << " : " << std::hexfloat << d << std::endl;
    std::cout << "f: " << std::setprecision(20) << std::scientific << f << " : " << std::hexfloat << f << std::endl;
    std::cout << "d-f: " << std::setprecision(20) << std::scientific << diff << " : " << std::hexfloat << diff << std::endl;
```
output:
```
d: 1.66666666666666657415e-01 : 0x1.5555555555555p-3
f: 1.66666671633720397949e-01 : 0x1.555556p-3
d-f: -4.96705374053441062188e-09 : -0x1.5555556p-28
```
(the first number is a decimal representation, the second is a hexadecimal representation, that shows exact content of the fp number)

Comment (katarinagresova) [reply geometry/mesh.cpp]: Thank you for comment and explanation. I have changed it to double constants.

**Review by vaclavblazek — APPROVED** [review]
seems ok

**Pull request merged**